agent/xds/builtinextensions/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,167 @@
+package ratelimit
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	envoy_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	envoy_route_v3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	envoy_ratelimit "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/local_ratelimit/v3"
+	envoy_http_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	envoy_type_v3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	envoy_resource_v3 "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/hashicorp/go-multierror"
+	"github.com/mitchellh/mapstructure"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	"github.com/hashicorp/consul/agent/xds/builtinextensiontemplate"
+	"github.com/hashicorp/consul/agent/xds/xdscommon"
+	"github.com/hashicorp/consul/api"
+)
+
+type ratelimit struct {
+	ProxyType     string
+	Listener      string
+	MaxTokens     int
+	TokensPerFill int
+	FillInterval  int
```

Comment:
nit: since `ratelimit` is not exported, I don't think we need the fields capitalized

Reply:
:thinking: I believe that the fields must be exported in order to use `mapstructure.Decode` on this struct: https://github.com/hashicorp/consul/blob/33f6edef2a1fbd40bae2026f18e68f4071985c3f/agent/xds/builtinextensions/ratelimit/ratelimit.go#L45

Reply:
Ah yes, thats for pointing that out!